decidim-core/lib/decidim/has_private_users.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Decidim
+  # A concern with the features needed when you want a model to be able to create
+  # private users
+  module HasPrivateUsers
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :participatory_space_private_users, class_name: "Decidim::ParticipatorySpacePrivateUser", as: :privatable_to, dependent: :destroy
+      has_many :users, through: :participatory_space_private_users, class_name: "Decidim::User", foreign_key: "private_user_to_id"
+
+      scope :visible_for, lambda { |user|
+                            joins("LEFT JOIN decidim_participatory_space_private_users ON
+                            decidim_participatory_space_private_users.privatable_to_id = #{table_name}.id")
+                              .where("(private_space = ? and decidim_participatory_space_private_users.decidim_user_id = ?) or private_space = ?", true, user, false)
+                          }
+
+      def self.non_private_spaces
```

Comment:
This should override `public_spaces` instead, using `super`:

```ruby
def self.public_spaces
  super.where(private_space: false)
end
```

Reply:
but, `public_spaces` is when spaces are `published` and then `non_private_spaces` when this `public_spaces` aren't private,  understand that this is different. If you think it's the same, I do what you say.

Reply:
Well, we aren't actually using `non_private_spaces` for anything else than this, so I don't think it deserves an actual method. From the consumer's point of view, `public_spaces` should be all the spaces that can be seen by an anonymous user, so I think this makes sense.